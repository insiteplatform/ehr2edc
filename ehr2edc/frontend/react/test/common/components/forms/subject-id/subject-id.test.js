import React from 'react';
import SubjectIdField from "../../../../../src/common/components/forms/subject-id/subject-id";

describe("Rendering the SubjectId input", () => {
    test("Renders without availableSubjectIds", () => {
        const field = shallow(<SubjectIdField/>);

        expect(field).toMatchSnapshot();
        expect(field.find("AutocompleteInput")).toHaveLength(0);
        expect(field.find("input[name='subjectIdentifier']")).toHaveLength(1);
    });

    test("Renders with availableSubjectIds", () => {
        const field = shallow(<SubjectIdField availableSubjects={{
            fromEDC: true,
            subjectIds: ["SUB-1", "SUB-2"]
        }}/>);

        expect(field).toMatchSnapshot();
        expect(field.find("AutocompleteInput")).toHaveLength(1);
        expect(field.find("input[name='subjectIdentifier']")).toHaveLength(0);
    });

    test("Renders disabled field with availableSubjectIds", () => {
        const field = shallow(<SubjectIdField availableSubjects={{
            fromEDC: true,
            subjectIds: ["SUB-1", "SUB-2"]
        }} disabled={true}/>);

        expect(field).toMatchSnapshot();
        expect(field.find("AutocompleteInput")).toHaveLength(0);
        expect(field.find("input[name='subjectIdentifier']")).toHaveLength(1);
    })
});

describe("Interaction with parent component", () => {
    const changedId = "changedId";
    test("Callback is triggered onChange", () => {
        const callback = jest.fn();
        const field = shallow(<SubjectIdField onChange={callback}/>);

        field.instance().subjectIdChange(changedId);

        expect(callback).toHaveBeenCalledWith(changedId);
    });
    test("When the subject id field has changed, the value gets updated", () => {
        const callback = jest.fn();
        const field = shallow(<SubjectIdField onChange={callback}/>);

        field.find(".subject-identifier-field").simulate("change", {target: {value: changedId}});

        expect(callback).toHaveBeenCalledWith(changedId);
    });
    test("When the subject id field has lost focus, the value gets trimmed", () => {
        const callback = jest.fn();
        const field = shallow(<SubjectIdField onChange={callback}/>);

        field.find(".subject-identifier-field").simulate("blur", {target: {value: ` ${changedId}\t`}});

        expect(callback).toHaveBeenCalledWith(changedId);
    });
});