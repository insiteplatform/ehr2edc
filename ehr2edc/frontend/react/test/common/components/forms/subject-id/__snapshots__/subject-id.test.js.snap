// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the SubjectId input Renders disabled field with availableSubjectIds 1`] = `
<input
  autoFocus={false}
  className="subject-identifier-field"
  disabled={true}
  name="subjectIdentifier"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Subject identifier"
  required={true}
  type="text"
  value=""
/>
`;

exports[`Rendering the SubjectId input Renders with availableSubjectIds 1`] = `
<AutocompleteInput
  alwaysRender={false}
  inputProps={
    Object {
      "autoFocus": false,
      "className": "subject-identifier-field",
    }
  }
  menuStyle={Object {}}
  onSelectionChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "label": "SUB-1",
        "value": "SUB-1",
      },
      Object {
        "label": "SUB-2",
        "value": "SUB-2",
      },
    ]
  }
  placeholder="Select EDC ID"
/>
`;

exports[`Rendering the SubjectId input Renders without availableSubjectIds 1`] = `
<input
  autoFocus={false}
  className="subject-identifier-field"
  disabled={false}
  name="subjectIdentifier"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Subject identifier"
  required={true}
  type="text"
  value=""
/>
`;
